### ISSUE TITLE
Unable to specify listen port with any IPv6 address

### ISSUE THREAD

**Issue by s1lversurfer** [hashicorp/boundary#701]
**Describe the bug**
Boundary will throw an error with any listen directive, when an IPv6 address and port are specified.
This happens with runtime options as well as when using the config file.

> Error initializing listener of type tcp: listen tcp: address :::8080: too many colons in address

**To Reproduce**
```
boundary dev -api-listen-address="[2001:db8::1:cee:c0de:b0b]:8080"
```
OR
```
listener "tcp" {
  address = "[2001:db8::1:cee:c0de:b0b]:8080"
  purpose = "api"
}
```
both result in:
> Error initializing listener of type tcp: listen tcp: address 2001:db8::1:cee:c0de:b0b:8080: too many colons in address

**Expected behavior**
Boundary should open a TCP socket on "2001:db8::1:cee:c0de:b0b" with port 8080.

**Comment by jefferai** [comment]
Ugh. I could have sworn this should work and that I'd tested it but, nope, it doesn't, and I guess I didn't. I'll be fixed in the next release. It's a (really, really annoying) mismatch between Go's net.SplitHostPort function which removes `[]` brackets from IPv6 addresses and its Listen function which requires them.

**Comment by jefferai** [comment]
If you're able to test that ^ then great! Otherwise it should work and will appear in 0.1.1.

**Comment by s1lversurfer** [comment]
Works for me.
Seems like you fixed both my issues overnight, thanks @jefferai !

**Comment by jefferai** [comment]
Note: I've merged that PR into #705 ; marking this as closed, when that is merged it will fix this too.
